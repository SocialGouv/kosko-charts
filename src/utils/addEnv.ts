import { assert } from "@sindresorhus/is";
import { Deployment } from "kubernetes-models/apps/v1/Deployment";
import { EnvVar } from "kubernetes-models/v1/EnvVar";

interface AddToContainerParams {
  deployment: Deployment;
  data: EnvVar;
  containerIndex?: number;
}

export const addEnv = ({
  deployment,
  data,
  containerIndex = 0,
}: AddToContainerParams): void => {
  assert.object(deployment.spec);
  assert.object(deployment.spec.template.spec);
  const container = deployment.spec.template.spec.containers[containerIndex];
  container.env = container.env ?? [];
  container.env.push(data);
};
