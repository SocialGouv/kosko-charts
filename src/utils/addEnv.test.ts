import assert from "assert";

import { addEnv } from "./addEnv";
import { Deployment } from "kubernetes-models/apps/v1/Deployment";
import { EnvVar } from "kubernetes-models/v1/EnvVar";

test("should assert FOO variable in env", () => {
  const deployment = new Deployment({
    metadata: { name: "test1" },
    spec: {
      selector: {},
      template: { spec: { containers: [{ name: "container1" }] } },
    },
  });
  addEnv({ deployment, data: new EnvVar({ name: "HELLO", value: "world" }) });
  expect(deployment).toMatchSnapshot();
});
