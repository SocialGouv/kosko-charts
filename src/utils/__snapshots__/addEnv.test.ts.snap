// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should assert FOO variable in env 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "metadata": Object {
    "name": "test1",
  },
  "spec": Object {
    "selector": Object {},
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "env": Array [
              Object {
                "name": "HELLO",
                "value": "world",
              },
            ],
            "name": "container1",
          },
        ],
      },
    },
  },
}
`;
